secrets/drivertest/drivertest.go

```diff
@@ -238,8 +238,39 @@ func testDecryptMalformedError(t *testing.T, newHarness HarnessMaker) {
 	}
 	keeper := secrets.NewKeeper(drv)
 
-	if _, err := keeper.Decrypt(ctx, []byte("malformed cipher message")); err == nil {
-		t.Error("Got nil, want decrypt error")
+	msg := []byte("I'm a secret message!")
+	encryptedMsg, err := keeper.Encrypt(ctx, msg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l := len(encryptedMsg)
+	for _, tc := range []struct {
+		name      string
+		malformed []byte
+	}{
+		{
+			name:      "wrong first byte",
+			malformed: append([]byte{encryptedMsg[0] + 1}, encryptedMsg[1:]...),
+		},
+		{
+			name:      "missing second byte",
+			malformed: append(encryptedMsg[:1], encryptedMsg[2:]...),
```

Comment:
This removes the byte for the rest of the test, though. Each of these slices should operate on a copy of `encryptedMsg`. The most concise way I can think of to do the copy is by creating a small helper function above:

```go
encryptedMsg, err := keeper.Encrypt(ctx, msg)
if err != nil {
	t.Fatal(err)
}
copyEncryptedMsg := func() []byte {
  return append([]byte(nil), encryptedMsg...)
}
```

Reply:
Done.